os.hpp

```diff
@@ -24,33 +24,46 @@
 
 // Define the user struct
 enum userID {sys, u1, u2};
-struct User {
-  userID id;
-  bool running;
+userID nextUser(userID user) {
+  switch(user) {
+    case sys: return u1; break;
+    case u1: return u2; break;
+    case u2: return sys; break;
+  }
+  return user;
+}
+struct Process {
+  int id;
+  int status;
   registers regs;
+  // list<int>* tlb;
 };
 
-extern queue<User> readyQueue;
-extern queue<User> blockedQueue;
-extern User currentUser;
+extern queue<Process> readyQueue;
+extern queue<Process> blockedQueue;
+extern Process currentProcess;
 
 // Declare semaphor as enum (locked/unlocked) and READY and BLOCKED constants
 enum e_semaphor {locked, unlocked} semaphor;
 #define READY 1
 #define BLOCKED 0
+#define RUNNING 2
+#define NONE 3
 
 // Variables used by the scheduler
 extern int sysclock;
 extern int switchTime;
+extern userID currentUser;
+extern Process processTable[64];
```

Comment:
I think we should consider not implementing a process table for now for simplicity's sake, since we've only got one process per user. Later we can modify it if we need to implement multiple processes per user.